google/cloud/storage/doc/storage-main.dox

```diff
@@ -0,0 +1,97 @@
+/*!
+
+@mainpage Google Cloud Storage C++ Client Library
+
+These pages contain the reference guide to the Google Cloud Storage (GCS)
+C++ Client Library.
+
+## Quickstart
+
+The following instructions show you how to perform basic tasks in Cloud Storage
+using the C++ client library.
+
+Costs that you incur in Cloud Storage are based on the resources you use. This
+quickstart typically uses less than $0.01USD worth of Cloud Storage resources.
+
+If you have never used Cloud Storage we recommend that you first follow one of
+the [language neutral quickstart guides][quickstart-link] before you follow this
+guide.
+
+### Before you begin
+
+1. Select or create a Google Cloud Platform (GCP) project.
+
+    [GO TO THE MANAGE RESOURCES PAGE][resource-link]
+
+    Make a note of the project id, you will need to use it later.
+
+2. Make sure that billing is enabled for your project
+
+    [LEARN HOW TO ENABLE BILLING][billing-link]
+
+3. Learn about [key terms and concepts][concepts-link] for Cloud Storage.
+
+### Downloading and Compiling the C++ Client Library
+
+The [source code][source-link] for the Cloud Storage C++ Client
+Library can be found on [GitHub][source-link]. The top-level README
+file in this repository includes detailed instructions on how to compile the
+library. The examples used in this guide should be automatically compiled when
+you follow said instructions.
```

Comment:
Add a section on authentication using implicit credentials defined by `GOOGLE_APPLICATION_CREDENTIALS` and setting default Google Cloud project defined by `GOOGLE_CLOUD_PROJECT`.